metricbeat/module/kafka/partition/partition.go

```diff
@@ -322,3 +320,111 @@ func checkRetryQuery(err error) (retry, reconnect bool) {
 
 	return false, false
 }
+
+func findMatchingBroker(
+	addr string,
+	advertised string,
+	brokers []*sarama.Broker,
+) *sarama.Broker {
+	// compare connection address to list of broker addresses
+	for _, b := range brokers {
+		if advertised != "" && b.Addr() == advertised {
+			return b
+		}
+
+		if b.Addr() == addr {
```

Comment:
We pass `b.Addr()` of our broker we selected to `findMatchingBroker`. So is the `b.Addr()` from our broker the one we configured or the one fetched from the meta data? I assume it is the one we configured, as otherwise it would always match with at least on the brokers or otherwise the broker connection would not work.